pxr/usdImaging/usdImagingGL/CMakeLists.txt

```diff
@@ -142,3 +146,4445 @@ pxr_build_test(testUsdImagingGLSurfaceShader
+# between idiff -p vs perceptualdiff
+pxr_register_test(testUsdImagingGLInheritedPurpose_none
+    COMMAND "${CMAKE_INSTALL_PREFIX}/tests/testUsdImagingGLBasicDrawing -guidesPurpose hide -renderPurpose hide -proxyPurpose hide -offscreen -lighting -frameAll -stage InheritedPurposes.usd -write testUsdImagingGLInheritedPurpose_none.png"
+    IMAGE_DIFF_COMPARE
+        testUsdImagingGLInheritedPurpose_none.png
+    FAIL 1
+    FAIL_PERCENT 2
+    PERCEPTUAL
+    EXPECTED_RETURN_CODE 0
+    TESTENV testUsdImagingGLInheritedPurpose
+)
+
+
+pxr_register_test(testUsdImagingGLInheritedPurpose_guides
+    COMMAND "${CMAKE_INSTALL_PREFIX}/tests/testUsdImagingGLBasicDrawing -guidesPurpose show -renderPurpose hide -proxyPurpose hide -offscreen -lighting -frameAll -stage InheritedPurposes.usd -write testUsdImagingGLInheritedPurpose_guides.png"
+    IMAGE_DIFF_COMPARE
+        testUsdImagingGLInheritedPurpose_guides.png
+    FAIL 1
+    FAIL_PERCENT 2
+    PERCEPTUAL
+    EXPECTED_RETURN_CODE 0
+    TESTENV testUsdImagingGLInheritedPurpose
+)
+
+
+pxr_register_test(testUsdImagingGLInheritedPurpose_render
+    COMMAND "${CMAKE_INSTALL_PREFIX}/tests/testUsdImagingGLBasicDrawing -guidesPurpose hide -renderPurpose show -proxyPurpose hide -offscreen -lighting -frameAll -stage InheritedPurposes.usd -write testUsdImagingGLInheritedPurpose_render.png"
+    IMAGE_DIFF_COMPARE
+        testUsdImagingGLInheritedPurpose_render.png
+    FAIL 1
+    FAIL_PERCENT 2
+    PERCEPTUAL
+    EXPECTED_RETURN_CODE 0
+    TESTENV testUsdImagingGLInheritedPurpose
+)
+
+
+pxr_register_test(testUsdImagingGLInheritedPurpose_proxy
+    COMMAND "${CMAKE_INSTALL_PREFIX}/tests/testUsdImagingGLBasicDrawing -guidesPurpose hide -renderPurpose hide -proxyPurpose show -offscreen -lighting -frameAll -stage InheritedPurposes.usd -write testUsdImagingGLInheritedPurpose_proxy.png"
+    IMAGE_DIFF_COMPARE
+        testUsdImagingGLInheritedPurpose_proxy.png
+    FAIL 1
+    FAIL_PERCENT 1
+    PERCEPTUAL
+    EXPECTED_RETURN_CODE 0
+    TESTENV testUsdImagingGLInheritedPurpose
+)
+
+
+pxr_register_test(testUsdImagingGLInheritedPurpose_all
+    COMMAND "${CMAKE_INSTALL_PREFIX}/tests/testUsdImagingGLBasicDrawing -guidesPurpose show -renderPurpose show -proxyPurpose show -offscreen -lighting -frameAll -stage InheritedPurposes.usd -write testUsdImagingGLInheritedPurpose_all.png"
+    IMAGE_DIFF_COMPARE
+        testUsdImagingGLInheritedPurpose_all.png
+    FAIL 1
+    FAIL_PERCENT 1
+    PERCEPTUAL
+    EXPECTED_RETURN_CODE 0
+    TESTENV testUsdImagingGLInheritedPurpose
+)
+
+
+pxr_register_test(testUsdImagingGLInheritedPurpose_Instanced_none
+    COMMAND "${CMAKE_INSTALL_PREFIX}/tests/testUsdImagingGLBasicDrawing -guidesPurpose hide -renderPurpose hide -proxyPurpose hide -offscreen -lighting -frameAll -stage InheritedPurposes_Instanced.usd -write testUsdImagingGLInheritedPurpose_Instanced_none.png"
+    IMAGE_DIFF_COMPARE
+        testUsdImagingGLInheritedPurpose_Instanced_none.png
```

Comment:
I think something is up with these testUsdImagingGLInheritedPurpose_Instanced_* baseline images. Although the tests all pass when I run them, if I actually look at the generated images vs. the baseline images, the framing appears to be very different. Can you confirm this on your side?

idiff reports that something like 1-2% of the pixels fail the perceptualdiff, so I think the fail threshold here might be too high.

For reference, we'd expect the baselines for the instanced tests to match the baselines for the non-instanced tests.

Reply:
I switched the instanced tests to match the baselines of the non-instanced tests and *_Instanced_guides, *_Instanced_render and *_Instanced_none still have 1-2% of the pixels fail. For me, I didn't see any framing differences and the output difference image reflected that.

I'm going to get rid of the *_Instanced_* baseline images and use the non-instanced ones. These three tests are failing for me with a `FAIL_PERCENT` of 1 so I'm not sure what we can do without bumping that value up to 1.5 or 2. Thoughts?

Reply:
Interestingly enough, when I run idiff on my test machine now I get an exact match between baseline and generated images. Out of curiosity what platform are you testing on?

In any case, I think your initial settings for `FAIL_PERCENT` on these tests are fine. We'll probably need to be tweaking a bunch of these as we go.